indexer/src/main/java/au/org/aodn/esindexer/service/StacCollectionMapperService.java

```diff
@@ -660,9 +585,34 @@ List<ContactsModel> mapContacts(MDMetadataType source) {
                 });
             }
         }
+
+        // get metadata contact
+        var mdContacts = MapperUtils.findMDContact(source);
+        if (!mdContacts.isEmpty()) {
+            for (var mdContact : mdContacts) {
+                var responsibilityValue = mdContact.getAbstractResponsibility().getValue();
+                if (!(responsibilityValue instanceof final CIResponsibilityType2 ciResponsibility)) {
+                    continue;
+                }
+                if (ciResponsibility.getParty() == null) {
+                    continue;
+                }
```

Comment:
can this be in a single block with `OR`?

Reply:
yes single block will be like this:
```java
                if (
                        !(responsibilityValue instanceof final CIResponsibilityType2 ciResponsibility)
                        || ciResponsibility.getParty() == null) {
                    continue;
                }
```

I just reckon, the condition statements are not very simple and maybe separating them is better for debugging and readability?

Reply:
if you put a breaker ( :red_circle: ) when debugging, that shouldn't be a problem, the IDE will show everything related.

Reply:
I didn't mean one block cannot debug, just for clearer reason. these two condition statements are not very simple & short, so multi-lines if conditions will be there. So I separated them into two parts for clearer debugging and reading. But i am ok for both version. I can change it soon.

Reply:
it's up to you, I ticked "Comment" option for this second review, instead of "Request changes"

![image](https://github.com/aodn/es-indexer/assets/26201635/c643a5f4-5a48-4bf2-99d1-0d9b735f89e0)

Reply:
Thank you. The next commit will have necessary changes 😃